### PR TITLE
Refactor refresh_indicator.1.dart to not use shrinkwrap

### LINKED ISSUE #116237: Don't use shrink wrap in our samples
https://github.com/flutter/flutter/blob/fa063eb4ce4140dec9a887c3130239c68f7b713e/examples/api/lib/material/refresh_indicator/refresh_indicator.1.dart uses shrinkWrap, but really it shouldn't. Unfortunately, it is currently hard to achieve this effect without shrinkWrapping. We should implement whatever additional Slivers are needed to not require shrinkWrapping here (e.g. SliverDecoration).

### PR DESCRIPTION
**Pull request by thkim1011**
This PR changes the example app into a custom scrollview with three slivers. The middle sliver has a nested scrollview of height 300 and only this nested sliver can trigger the refresh indicator.

Fixes https://github.com/flutter/flutter/issues/116237.

### CODE REVIEW

File: examples/api/lib/material/refresh_indicator/refresh_indicator.1.dart [hunk 1]
```diff
@@ -47,10 +47,10 @@ class RefreshIndicatorExample extends StatelessWidget {
         notificationPredicate: (ScrollNotification notification) {
```
Comment (Piinks) [inline examples/api/lib/material/refresh_indicator/refresh_indicator.1.dart:47]: There is a comment just above this, can you adjust it to

```dart
// By default this is set to `notification.depth == 0`, which ensures
// the only the scroll notifications from the first scroll view are listened to.
```

**Review by Piinks — COMMENTED** [review]
Since most folks view this on the web in a desktop environment, they wont actually be able to pull it down as it says. Can you add in MaterialApp for scrollBehavior: `MaterialScrollBehavior().copyWith(dragDevices: PointerDeviceKind.values.toSet())`? That way no matter where folks are checking out this example, they will be able to drag on it. :)

**Review by Piinks — APPROVED** [review]
LGTM - this will need a test exemption from Hixie though.

**Comment by Hixie** [comment]
test-exempt: code refactor with no semantic change

**Pull request merged**